From 170c5323c8e9d8307e817739ac6eff1983b3578b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1=C5=A1=20Hur=C3=BDn?= <huryn@2n.com>
Date: Thu, 7 May 2026 14:50:30 +0200
Subject: [PATCH] Fix voltage value on AOD screen

---
 .../systemui/statusbar/KeyguardIndicationController.java    | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/KeyguardIndicationController.java b/packages/SystemUI/src/com/android/systemui/statusbar/KeyguardIndicationController.java
index ce8178920f491..de19d4f4b3303 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/KeyguardIndicationController.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/KeyguardIndicationController.java
@@ -1764,9 +1764,9 @@ private float getRealtimeChargingVoltage(
             }
 
             final float voltageMicroVolts = voltageMilliVolts * 1000f;
-            return voltageMicroVolts >= 3_000_000f && voltageMicroVolts <= 6_000_000f
-                    ? voltageMicroVolts
-                    : fallbackVoltageMicroVolts;
+            return fallbackVoltageMicroVolts > 0
+                    ? fallbackVoltageMicroVolts
+                    : voltageMicroVolts;
         }
 
         private float getRealtimeChargingWattage(
