From 6475b92097831d256eaf2a49cf40f515654bdf65 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1=C5=A1=20Hur=C3=BDn?= <huryn@2n.com>
Date: Wed, 13 May 2026 07:48:50 +0200
Subject: [PATCH] Smooth brithenss control in AOD

---
 .../systemui/doze/DozeScreenBrightness.java        | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/doze/DozeScreenBrightness.java b/packages/SystemUI/src/com/android/systemui/doze/DozeScreenBrightness.java
index 73be0034dbef5..adfd0fcc439f5 100644
--- a/packages/SystemUI/src/com/android/systemui/doze/DozeScreenBrightness.java
+++ b/packages/SystemUI/src/com/android/systemui/doze/DozeScreenBrightness.java
@@ -361,11 +361,23 @@ private int computeWallpaperScrimOpacity(int sensorValue) {
         return mSensorToWallpaperScrimOpacity[sensorValue];
     }
 
+    private float mAverageSensorValue = -1f;
+
+    // Rychlost změny. 0.2f je dobrý začátek (reaguje rozumně rychle, ale žehlí šum)
+    private static final float ALPHA = 0.2f;
+
     private float computeBrightness(int sensorValue) {
         if (sensorValue < 0) {
             return -1;
         }
-        return mSensorToBrightness[sensorValue];
+        if (mAverageSensorValue == -1f) {
+            mAverageSensorValue = sensorValue;
+        } else {
+            mAverageSensorValue = mAverageSensorValue + ALPHA * (sensorValue - mAverageSensorValue);
+        }
+        int averagedSensorValue = Math.round(mAverageSensorValue);
+        averagedSensorValue = Math.max(0, Math.min(255, averagedSensorValue));
+        return mSensorToBrightness[averagedSensorValue];
     }
 
     @Override
