From 9d661a537307085ebaaf35fa5f1d1de46f828240 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1=C5=A1=20Hur=C3=BDn?= <huryn@2n.com>
Date: Tue, 20 Jan 2026 19:34:57 +0100
Subject: [PATCH] Fixed Doze screen brightness

---
 .../systemui/doze/DozeScreenBrightness.java   | 21 ++++++++++++-------
 1 file changed, 13 insertions(+), 8 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/doze/DozeScreenBrightness.java b/packages/SystemUI/src/com/android/systemui/doze/DozeScreenBrightness.java
index 2f71179097c07..73be0034dbef5 100644
--- a/packages/SystemUI/src/com/android/systemui/doze/DozeScreenBrightness.java
+++ b/packages/SystemUI/src/com/android/systemui/doze/DozeScreenBrightness.java
@@ -276,8 +276,8 @@ public void onSensorChanged(SensorEvent event) {
 
     public void updateBrightnessAndReady(boolean force) {
         if (force || mRegistered || mDebugBrightnessBucket != -1) {
-            int sensorValue = mDebugBrightnessBucket == -1
-                    ? mLastSensorValue : mDebugBrightnessBucket;
+            int sensorValue = normalizeSensorValue(mDebugBrightnessBucket == -1
+                    ? mLastSensorValue : mDebugBrightnessBucket);
             boolean brightnessReady;
             float brightness = computeBrightness(sensorValue);
             brightnessReady = brightness >= 0;
@@ -306,6 +306,16 @@ public void updateBrightnessAndReady(boolean force) {
         }
     }
 
+    public static int normalizeSensorValue(int rawSensorValue) {
+        if (rawSensorValue > 15000) rawSensorValue = 15000;
+        if (rawSensorValue < 0) rawSensorValue = 0;
+
+        double normalized = rawSensorValue / 15000.0;
+        double curve = Math.sqrt(normalized);
+
+        return (int) Math.round(curve * 255);
+    }
+
     private boolean lightSensorSupportsCurrentPosture() {
         return mLightSensorOptional != null
                 && mDevicePosture < mLightSensorOptional.length;
@@ -355,12 +365,7 @@ private float computeBrightness(int sensorValue) {
         if (sensorValue < 0) {
             return -1;
         }
-        for (int i = 0; i < mSensorToBrightness.length; i++) {
-            if (sensorValue < mSensorToBrightness[i]) {
-                return i;
-            }
-        }
-        return 255;
+        return mSensorToBrightness[sensorValue];
     }
 
     @Override
